src/praisonai-code/praisonai_code/tool_resolver.py

```
@@ -0,0 +1,1241 @@
+        if self._praisonai_tools_available is None:
+            from praisonai_code._framework_availability import is_available
+            self._praisonai_tools_available = is_available("praisonai_tools")
+
+        if self._praisonai_tools_available:
+            try:
+                from praisonai_tools import __all__ as praisonai_tools_all
+                for name in praisonai_tools_all:
+                    if name not in available:
+                        available[name] = ("External tool from praisonai-tools", "external")
+            except Exception:
+                pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)